### PR TITLE
Fixes #6633: Smart variables and smart class parameters can not have the same parameter name

### PR DESCRIPTION
**Pull request by orrabin**
... same parameter name

### CODE REVIEW

File: app/views/lookup_keys/_fields.html.erb [hunk 1]
```diff
@@ -1,5 +1,4 @@
-
-<div id='<%= (f.object.key || 'new_lookup_keys').to_s.gsub(' ','_')%>' class='tab-pane fields' >
+<div id='<%= (f.object.key || 'new_lookup_keys').to_s.gsub(' ','_') + '_' + f.object.id.to_s%>' class='tab-pane fields' >
```
Comment (orrabin) [inline app/views/lookup_keys/_fields.html.erb]: The css id's relied on the `f.object.key` being unique. 
When i changed the uniqueness to scoped uniqueness there could two identical keys (one for a smart variable and one for a smart class parameter) and that created a problem when trying to delete one of them. 
I added the `object.id` so the css id would be unique.

**Comment by lzap** [comment]
Ok reproduced, the error is:

```
has already been taken
```

with your patch it does save. I haven't tested if the parameter applies really tho. @ohadlevy do I need to test this? Other than that :+1:

File: app/models/lookup_key.rb [hunk 2]
```diff
@@ -31,8 +31,9 @@ def audit_class
 
   before_validation :validate_and_cast_default_value
 
-  validates :key, :uniqueness => true, :unless => Proc.new{|p| p.is_param?}
+  validates :key, :uniqueness => {scope: :is_param}, :unless => Proc.new{|p| p.is_param?}
```
Comment (GregSutcliffe) [inline app/models/lookup_key.rb]: This is causing failed tests, see the Jenkins output for details.

Comment (ohadlevy) [inline app/models/lookup_key.rb]: its a ruby 1.9 syntax, can you change to 

``` ruby
{:scope => is_param }
```

Comment (isratrade) [inline app/models/lookup_key.rb]: @orrabin, you also need to add a colon to make `:is_param` a symbol 

``` ruby
{:scope => :is_param }
```

**Comment by orrabin** [comment]
@GregSutcliffe thanks, I changed the scope syntax

**Comment by orrabin** [comment]
[test] again

**Comment by lzap** [comment]
Review done guys? Come on, we don't introduce new stale PRs! :-)

**Comment by domcleal** [comment]
I think a unit test would be good for this, to confirm the behaviour that you can have two parameters in their separate scopes, but otherwise looks good.

File: app/views/lookup_keys/_fields.html.erb (hunk 1 above)
Comment (dLobatog) [inline app/views/lookup_keys/_fields.html.erb]: That should be fine,  `f.object.id.to_s%>` lacks a white space between the `s` and the `%`

**Comment by orrabin** [comment]
@domcleal I added unit tests

**Comment by domcleal** [comment]
Merged as e6e9f15 for Foreman 1.6.0.  Thanks @orrabin!

(Note, I amended the commit message to say what the _new_ behaviour is, this is preferable.)

**Pull request closed without merging**